kymatio/scattering1d/scattering1d.py

```diff
@@ -355,11 +355,17 @@ def forward(self, x):
             is `False`, it is a dictionary indexed by tuples of filter indices.
         """
         # basic checking, should be improved
-        if len(x.shape) != 2:
```

Comment:
I prefer `len(x.shape)` because `.dim` is very pytorch-bound (the numpy equivalent is `.ndim`)